cpp/src/randomforest/randomforest.cu

```diff
@@ -341,6 +343,126 @@ std::vector<unsigned char> save_model(ModelHandle model) {
+           "Error! mismatch in the position of the node between concatenated "
+           "forest and the"
+           " individual forests ");
+    ASSERT(node_from_concat.cleft() == node_from_indiv.cleft(),
+           "Error! mismatch in the position of the node between concatenated "
+           "forest and the"
+           " individual forests ");
+    ASSERT(
+      node_from_concat.split_index() == node_from_indiv.split_index(),
+      "Error! split index value mismatch between concatenated forest and the"
+      " individual forests ");
+  }
+}
+
+/**
+ * @brief Compares the concatenated treelite model with the information of the forest
+ *   present in the different workers. If there is a difference in the two then an error
+ *   statement will be thrown.
+ * @param[in] concat_tree_handle: ModelHandle for the concatenated forest.
+ * @param[in] treelite_handles: List containing ModelHandles for the forest present in
+ *   each worker.
+ */
+void compare_concat_forest_to_subforests(
+  ModelHandle concat_tree_handle, std::vector<ModelHandle> treelite_handles) {
+  size_t concat_forest;
+  size_t total_num_trees = 0;
+  for (int forest_idx = 0; forest_idx < treelite_handles.size(); forest_idx++) {
+    size_t num_trees_each_forest;
+    TREELITE_CHECK(TreeliteQueryNumTree(treelite_handles[forest_idx],
+                                        &num_trees_each_forest));
+    total_num_trees = total_num_trees + num_trees_each_forest;
+  }
+
+  TREELITE_CHECK(TreeliteQueryNumTree(concat_tree_handle, &concat_forest));
+
+  ASSERT(
+    concat_forest == total_num_trees,
+    "Error! the number of trees in the concatenated forest and the sum "
+    "of the trees present in the forests present in each worker are not equal");
+
+  int concat_mod_tree_num = 0;
+  tl::Model& concat_model = *(tl::Model*)(concat_tree_handle);
+  for (int forest_idx = 0; forest_idx < treelite_handles.size(); forest_idx++) {
+    tl::Model& model = *(tl::Model*)(treelite_handles[forest_idx]);
+
+    ASSERT(
+      concat_model.num_feature == model.num_feature,
+      "Error! number of features mismatch between concatenated forest and the"
+      " individual forests ");
+    ASSERT(concat_model.num_output_group == model.num_output_group,
+           "Error! number of output group mismatch between concatenated forest "
+           "and the"
+           " individual forests ");
+    ASSERT(concat_model.random_forest_flag == model.random_forest_flag,
+           "Error! random forest flag value mismatch between concatenated "
+           "forest and the"
+           " individual forests ");
+
+    for (int indiv_trees = 0; indiv_trees < model.trees.size(); indiv_trees++) {
+      compare_trees(concat_model.trees[concat_mod_tree_num + indiv_trees],
+                    model.trees[indiv_trees]);
+    }
+    concat_mod_tree_num = concat_mod_tree_num + model.trees.size();
+  }
+}
+
+/**
+ * @brief Concatenates the forest information present in different workers to
```

Comment:
Should specify clearly that this allocates and returns a new model that is owned by (and must be freed by) the caller.